### PR TITLE
Develop ex3

### PR DESCRIPTION
**Pull request by barakmaron**

### CODE REVIEW

File: src/ex3/scripts/CliApp.mjs [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+/* eslint-disable class-methods-use-this */
+/* eslint-disable camelcase */
+import inquirer from 'inquirer';
+import { createSpinner } from 'nanospinner';
+import figlet from 'figlet';
+import gradient from 'gradient-string';
+import chalk from 'chalk';
+import asciifyImage from 'asciify-image';
+import ItemManager from './ItemManager.mjs';
+import {
```
Comment (or-nuri-monday) [inline src/ex3/scripts/CliApp.mjs:11]: This is import is too long. You could import STRINGS and call each string that you need using STRINGS.XXX

File: src/ex3/scripts/CliApp.mjs [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+          CLI_HELP_INSTRUCTIONS_SECOND_SECTION}`,
+      },
+    };
+  }
+
+  /**
+    * init the tasks in items manager
+    */
+  async init() {
+    await this.item_manager.SetArrayFromFile();
```
Comment (or-nuri-monday) [inline src/ex3/scripts/CliApp.mjs:93]: You could call it from the constructor.

File: src/ex3/scripts/ItemManager.mjs [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    this.pokemons = [];
+    this.tasks = [];
+    this.file_manager = new FileManager();
+  }
+
+  /**
+     * init pokemon names array
+     */
+  async init() {
+    const pokemons_names_response = await this.pokemon_client.GetPokemonsNames();
```
Comment (or-nuri-monday) [inline src/ex3/scripts/ItemManager.mjs:20]: You could use it in the constructor

File: src/ex3/scripts/ItemManager.mjs [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+  /**
+     * gets string and checks what case it is
+     * @param {string} task_text string from input
+     */
+  async AddTask(task_text) {
+    // parse text to int
+    const pokemon_id = Number.parseInt(task_text, 10);
+    // check if text has commas
+    if (task_text.indexOf(',') > -1) {
```
Comment (or-nuri-monday) [inline src/ex3/scripts/ItemManager.mjs:46]: What is I inserted a regular task that contains a comma?

File: src/ex3/scripts/ItemManager.mjs [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    } else { this.InsertResultToArray(pokemon); }
+    this.file_manager.WriteToFileTasksArray(this.tasks);
+  }
+
+  /**
+     * removes a task from array
+     * @param {int} task_to_remove_id task location in array
+     */
+  RemoveTask(task_to_remove_id) {
+    const current_task_length = this.tasks.length;
```
Comment (or-nuri-monday) [inline src/ex3/scripts/ItemManager.mjs:75]: You could implement it in a more elegant way (filter for example).
It is not a common way to validate the id

File: src/ex3/scripts/ItemManager.mjs [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+  /**
+     * send id to pokemon client and insert it to tasks array
+     * @param {int} id
+     */
+  async GetOnePokemon(id) {
+    // send the id to pokemon client
+    const result = Promise.resolve(this.pokemon_client.GetPokemonById(id));
+    // wait for response
+    await result;
+    return result;
```
Comment (or-nuri-monday) [inline src/ex3/scripts/ItemManager.mjs:92]: You could write `return await this.pokemon_client.GetPokemonById(id)`

File: src/ex3/scripts/ItemManager.mjs [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+
+  /**
+     * parse string from input send it to pokemon client and insert it to tasks array
+     * @param {string} ids string from input
+     */
+  async GetMultiplePokemons(ids) {
+    const pokemon_ids = ids.split(',');
+    const result = Promise.resolve(this.pokemon_client.GetPokemonsByList(pokemon_ids));
+    await result;
+    return result;
```
Comment (or-nuri-monday) [inline src/ex3/scripts/ItemManager.mjs:103]: Same as above

File: src/ex3/scripts/ItemManager.mjs [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+  InsertResultToArray(result) {
+    // check if found pokemon
+    if (typeof result.name !== 'undefined' && result.name !== 'Error') {
+      if (!this.CheckIfPokemonExists(result.id)) {
+        this.tasks.push({
+          name: result.name, images: result.images, id: result.id, completed: false,
+        });
+      }// found pokemon add the name and id of pokemon
+    } else if (typeof result === 'object') {
+      this.tasks.push({ data: result.message, id: 'Error', completed: false });
```
Comment (or-nuri-monday) [inline src/ex3/scripts/ItemManager.mjs:120]: Why are we adding an error to the tasks arr?

File: src/ex3/scripts/ItemManager.mjs [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+     * parses response into pokemons array
+     * @param {Array} response
+     */
+  ParsePokemonNamesResponse(response) {
+    if (response.name !== 'Error') {
+      response.forEach((pokemon) => {
+      // split url string and get rid of empty elements
+        const parsed_url = pokemon.url.split('/').filter((item) => item);
+        const id = parsed_url[parsed_url.length - 1];
+        this.pokemons.push({ name: pokemon.name, id });
```
Comment (or-nuri-monday) [inline src/ex3/scripts/ItemManager.mjs:166]: Couldn't you get the name from the response itself and not from the URL?

Comment (barakmaron) [reply src/ex3/scripts/ItemManager.mjs:166]: the response to this API call is an array of objects 
{
"name": STRING,
"url": STERING
}
so I get the name from the object and the id by parsing the url filed

**Pull request merged**